recipes/cad1/task1/baseline/enhance.py

```diff
@@ -0,0 +1,309 @@
+""" Run the dummy enhancement. """
+import json
+import logging
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import hydra
+import numpy as np
+import pandas as pd
+import torch
+from omegaconf import DictConfig
+from scipy.io import wavfile
+from torchaudio.pipelines import HDEMUCS_HIGH_MUSDB
+from torchaudio.transforms import Fade
+from tqdm import tqdm
+
+from clarity.enhancer.compressor import Compressor
+from clarity.enhancer.nalr import NALR
+
+logger = logging.getLogger(__name__)
+
+
+def separate_sources(
+    model,
+    mix,
+    sample_rate,
+    segment=10.0,
+    overlap=0.1,
+    device=None,
+):
+    """
+    Apply model to a given mixture. Use fade, and add segments together in order to add model segment by segment.
+
+    Arguments
+    ----------
+        model (torch.nn.Module): model to use for separation
+        mix (torch.Tensor): mixture to separate, shape (batch, channels, time)
+        sample_rate (int): sampling rate of the mixture
+        segment (int): segment length in seconds
+        overlap (float): overlap between segments, between 0 and 1
+        device (torch.device, str, or None): if provided, device on which to
+            execute the computation, otherwise `mix.device` is assumed.
+            When `device` is different from `mix.device`, only local computations will
+            be on `device`, while the entire tracks will be stored on `mix.device`.
+
+    Returns
+        torch.Tensor: estimated sources
+
+    Based on https://pytorch.org/audio/main/tutorials/hybrid_demucs_tutorial.html
+    """
+    if device is None:
```

Comment:
Could simplify to the following without losing readability
```suggestion
    device = mix.device if device is None else torch.device(device)
```